projects/optic/src/config.ts

```diff
@@ -177,3 +163,31 @@ export const initializeRules = async (config: RawYmlConfig) => {
 
   config.ruleset = [...rulesetMap.values()];
 };
+
+export async function initializeConfig(): Promise<OpticCliConfig> {
```

Comment:
Moved the code from init into here, to be able to add authentication + client initalization here. CLIConfig might not be the correct word for this anymore though - could rename to something else

Reply:
Seems fine to me.